mango-viz/bdgenomics/mango/pileup/sources.py

```diff
@@ -79,6 +80,20 @@ def __init__(self, json):
         #: dictionary containing source elements (viz, source, sourceOptions, label)
         self.dict_ = json
 
+class DataFrameSource(Source):
+    
+    def __init__(self, dataframe):
+        """Initializes dataframe. Converts it to raw json. 
+
+        Args:
+            :param dataframe: dataframe
+        """
+        feature_transformed_json = parse_bed_dataframe(dataframe)
```

Comment:
I think the way @devin-petersohn set this up is that you should be able to call the _to_json function directly from the dataframe